asynchronous-data-flow/sycl-2.2/03_interacting_with_data_on_the_host.md

```diff
@@ -78,18 +78,51 @@ without blocking the execution.
     auto cgH = [=] (handler& h) {
       auto accA = bufA.get_access<access::mode::read>(h);
       h.parallel_for<class kernel>(range, SomeKernel(accA));
-      h.update_from_device(accA, hostPtr);
-      };
+      h.update_from_device(hostPtr, accA);
+    };
     qA.submit(cgH);
 ```
 
 Note that the users can query the event returned by the submit to check if the
 command group has finished (and, therefore, the host pointer has been updated).
 
+Additionally, users may want to manually update the device data with some host
+data.
+
+```cpp
+auto cgH = [=] (handler& h) {
+  auto accA = bufA.get_access<access::mode::read_write>(h);
+  h.update_to_device(accA, hostPtr);
+  h.parallel_for<class kernel>(range, SomeKernel(accA));
+};
+qA.submit(cgH);
+```
+
+#### Access restrictions
+
+There are a few restrictions on the access mode and target of the accessor
+supplied to `update_from_device` and `update_to_device`.
```

Comment:
The following restrictions apply to access mode and target of the accessor provided to...